packages/service-library/src/servicelib/background_task_utils.py

```diff
@@ -0,0 +1,50 @@
+import datetime
+import functools
+from collections.abc import Callable, Coroutine
+from typing import Any, ParamSpec, TypeVar
+
+from .background_task import periodic
+from .redis import RedisClientSDK, exclusive
+
+P = ParamSpec("P")
+R = TypeVar("R")
+
+
+def exclusive_periodic(
+    redis_client: RedisClientSDK | Callable[..., RedisClientSDK],
+    *,
+    task_interval: datetime.timedelta,
+    retry_after: datetime.timedelta = datetime.timedelta(seconds=1),
+) -> Callable[
+    [Callable[P, Coroutine[Any, Any, None]]], Callable[P, Coroutine[Any, Any, None]]
+]:
+    """decorates a function to become exclusive and periodic.
+
+    Arguments:
+        client -- The Redis client
+        task_interval -- the task interval, i.e. how often the task should run
+        retry_after -- in case the exclusive lock cannot be acquired or is lost, this is the retry interval
+
+    Raises:
+        Nothing
+
+    Returns:
+        Nothing, a periodic method does not return anything as it runs forever.
+    """
+
+    def _decorator(
+        func: Callable[P, Coroutine[Any, Any, None]],
+    ) -> Callable[P, Coroutine[Any, Any, None]]:
+        @periodic(interval=retry_after)
+        @exclusive(
+            redis_client,
+            lock_key=f"lock:exclusive_periodic_task:{func.__name__}",
```

Comment:
So the lock key suggest the exclusivity is per function. 
I am a bit concerned here. Look
```
>>> simcore_service_webserver.application.setup_login.__name__
'setup_login'
```

the `__name__` does not seem to include the namespaces. There are big chances we repeat function names

Reply:
true that I had forgotten about it...

Reply:
added the module in front